src/com/mendix/widget/PullToRefresh/handlers/PullToRefresh.ts

```diff
@@ -0,0 +1,215 @@
+            }
+        }
+    }
+
+    private onReset() {
+        const { cssProp, pullToRefreshElement, classPrefix } = this.settings;
+        domClass.remove(pullToRefreshElement.id, `${classPrefix}refresh`);
+        domStyle.set(pullToRefreshElement, cssProp, "0px");
+
+        this.state = "pending";
+    }
+
+    private onTouchStart(event: TouchEvent) {
+        if (this.state !== "pending") return;
+        if (this.isScrollActive(event.target as HTMLElement)) {
+            this.state = "scrolling";
+            return;
+        }
+
+        clearTimeout(this.timeout);
+
+        this.pullStart.screenX = event.touches[0].screenX;
+        this.pullStart.screenY = event.touches[0].screenY;
+        this.enable = this.settings.triggerElement.contains(event.target as Node);
+        this.state = "pending";
+        this.lastDistance = 0;
+        this.update();
+    }
+
+    private onTouchMove(event: TouchEvent) {
+        const { pullToRefreshElement, maximumDistance, thresholdDistance, cssProp, classPrefix } = this.settings;
+        const touch = event.touches[0];
+        const touchElement = document.elementFromPoint(touch.clientX, touch.clientY);
+        this.pullMoveY = event.touches[0].screenY;
+
+        if (!this.enable || this.state === "refreshing" || this.updating || this.state === "scrolling") return;
+
+        if (this.isScrollActive(touchElement as HTMLElement)) return;
+
+        if (this.state === "pending") {
+            domClass.add(pullToRefreshElement.id, `${classPrefix}pull`);
+            this.state = "pulling";
+            this.update();
+        }
+        if (this.pullStart.screenY && this.pullMoveY) {
+            this.distance = this.pullMoveY - this.pullStart.screenY;
+        }
+        if (this.distance > 0 && Math.abs(this.lastDistance - this.distance) > 1) {
+            this.lastDistance = this.distance;
+            event.preventDefault();
+            domStyle.set(pullToRefreshElement, cssProp, `${this.distanceResisted}px`);
+            this.distanceResisted = this.settings.resistanceFunction(this.distance / thresholdDistance)
+                * Math.min(maximumDistance, this.distance);
+
+            if (this.state === "pulling" && this.distanceResisted > thresholdDistance) {
+                domClass.add(pullToRefreshElement.id, `${classPrefix}release`);
+                this.state = "releasing";
+                this.update();
+            }
+
+            if (this.state === "releasing" && this.distanceResisted < thresholdDistance) {
+                domClass.remove(pullToRefreshElement.id, `${classPrefix}release`);
+                this.state = "pulling";
+                this.update();
+            }
+        }
+    }
+
+    private onTouchEnd() {
+        const { pullToRefreshElement, onRefresh, cssProp, classPrefix } = this.settings;
+
+        if (this.state === "releasing" && this.distanceResisted > this.settings.thresholdDistance) {
```

Comment:
Check for `this.distanceResisted > this.settings.thresholdDistance` not needed, `this.state` can only be ``releasing` when that condition is true, right?